### PR TITLE
Add reviews fe fp66

### PR DESCRIPTION
**Pull request by JosephOri**
partial pr

### CODE REVIEW

File: src/api/base.client.ts
Comment (nitzanto) [inline src/api/base.client.ts:1]: Try avoiding adding files that are not relevant to the PR. Because you didn't change anything here.

File: src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx [hunk 1]
```diff
@@ -20,31 +18,15 @@ const GuideCard = ({ guide }: Props) => {
           <Typography variant="h6">{guide.title}</Typography>
         </Box>
         <Typography variant="body2" color="textSecondary" mb={2}>
-          Issue: {guide.issue}
+          Created by User ID: {guide.creatorId}
         </Typography>
```
Comment (nitzanto) [inline src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx]: Why did you remove the guide issue? its relevant.
Also why displaying the user id and not username?

File: src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx [hunk 2]
```diff
@@ -20,31 +18,15 @@ const GuideCard = ({ guide }: Props) => {
...
-              {guide.dislikes}
-            </Typography>
-          </Box>
+        <Typography variant="body2" color="textSecondary" mb={2}>
+          Created at: {new Date(guide.createdAt).toLocaleDateString()}
+        </Typography>
+        <Box display="flex" alignItems="center" mt={2}>
+          <Typography variant="body2" color="textSecondary">
+            Total Stars: {guide.starsTotalSum}
+          </Typography>
```
Comment (nitzanto) [inline src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx]: Should be guide.rating not guide.starsTotalSum.

File: src/app/(admin)/new_dashboard/guides/components/GuideList/index.tsx [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
-  <Grid container spacing={2}>
-    {guides.map((guide, index) => (
-      <Grid item xs={12} sm={6} md={4} key={index}>
-        <GuideCard guide={guide} />
+const GuideList = ({ guideItems }: Props) => {
+  if (!guideItems) {
+    return (
+      <Grid item xs={12}>
+        <Typography>Loading...</Typography>
       </Grid>
```
Comment (nitzanto) [inline src/app/(admin)/new_dashboard/guides/components/GuideList/index.tsx]: Put the loader in the parent component.

File: src/app/(admin)/new_dashboard/guides/page.tsx [hunk 4]
```diff
@@ -1,26 +1,45 @@
...
+                  searchQuery={searchQuery}
+                  onSearchChange={handleSearchChange}
+                />
+              </Box>
+              <GuideList guideItems={filteredGuides} />
+            </>
+          )}
+          {isSuccess && guides.length === 0 && (
+            <Typography>No guides available</Typography>
+          )}
```
Comment (nitzanto) [inline src/app/(admin)/new_dashboard/guides/page.tsx]: Try adding loading skeletons from MUI. Or alternatively use a ciruclar loader placed at the center of the page.
Also why not to display the search bar? You can just disable it while there's no data or it won't even search anything while there's no data.

Add an alert component from MUI to display the error.

File: src/app/(admin)/new_dashboard/guides/page.tsx [hunk 5]
```diff
@@ -1,26 +1,45 @@
...
+
+  let guides: Guide[] = [];
+
+  if (isSuccess && "data" in guideItems) {
+    guides = guideItems.data;
+  }
+
+  const { searchQuery, filteredGuides, handleSearchChange } = useSearchGuides(
+    guides
+  );
```
Comment (nitzanto) [inline src/app/(admin)/new_dashboard/guides/page.tsx]: if(isSuccess) just set guideItems as a success response. Why would you redefine another variable its not necessary.

**Review by nitzanto — CHANGES_REQUESTED** [review]

File: src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx (hunk 2 above)
Comment (JosephOri) [reply src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx]: this will be fixed once we have the reviews array, than we will calculate the avrage rating by `starsTotalSum / reviews.length`

File: src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx (hunk 1 above)
Comment (JosephOri) [reply src/app/(admin)/new_dashboard/guides/components/GuideCard/index.tsx]: I can add issue also in the backend no problem, and we need a way to get the username other than making a request to get the user

File: src/app/(admin)/new_dashboard/guides/page.tsx (hunk 5 above)
Comment (JosephOri) [reply src/app/(admin)/new_dashboard/guides/page.tsx]: can you please explain about it?
